### DISCUSSION TITLE
ZoeDepth

### DISCUSSION THREAD

**Discussion by AugmentedRealityCat** [thygate/stable-diffusion-webui-depthmap-script#123] [Ideas]
Can the code from this new ZoeDepth monocular depth estimation research be adapted to work with the depth map extension ?

https://github.com/isl-org/ZoeDepth

**Comment by graemeniedermayer** [comment]
Just an example from zoeDepth (The contours are from downsampling to 8-bit their 16-bit seems was only outputting black images). I think you might have been onto something when you suggested ZoeDepth was the repo used in those other examples. Because I think this would be really high quality with 16-bit depth.

zoe_n (indoor version zoe_k is outdoor)
https://user-images.githubusercontent.com/38518120/222562501-ec684012-b7bb-4a17-9fec-e7f3e78929a6.mp4

The thing that seems really cool is they appear to be calibrated depth which should mean that depth scaling is no longer needed. Although you still need might need full scene scaling.

**Reply by AugmentedRealityCat** [reply]
Thanks for sharing this example. This is really good !

Could it be then that the [division by 255 I read about on Reddit](https://github.com/thygate/stable-diffusion-webui-depthmap-script/issues/125) was part of the ZoeDepth code?
And maybe that could explain the black output you got in 16 bit as well?

Time to use the code search function on that repo ! 
I really appreciate your help a lot by the way - it's easy to get lost in all that complexity :)

**Reply by AugmentedRealityCat** [reply]
Here are the lines I found that had a division by 255

https://github.com/isl-org/ZoeDepth/blob/5b481547085d6cc26888f613fea37eab77fd7c8a/zoedepth/data/vkitti2.py#L152-L155

https://github.com/isl-org/ZoeDepth/blob/5b481547085d6cc26888f613fea37eab77fd7c8a/zoedepth/data/ddad.py#L97

https://github.com/isl-org/ZoeDepth/blob/5b481547085d6cc26888f613fea37eab77fd7c8a/zoedepth/utils/misc.py#L148

https://github.com/isl-org/ZoeDepth/blob/5b481547085d6cc26888f613fea37eab77fd7c8a/zoedepth/utils/misc.py#L352

https://github.com/isl-org/ZoeDepth/blob/5b481547085d6cc26888f613fea37eab77fd7c8a/zoedepth/data/vkitti.py#L119

https://github.com/isl-org/ZoeDepth/blob/5b481547085d6cc26888f613fea37eab77fd7c8a/zoedepth/data/data_mono.py#L349

https://github.com/isl-org/ZoeDepth/blob/5b481547085d6cc26888f613fea37eab77fd7c8a/zoedepth/data/data_mono.py#L381

https://github.com/isl-org/ZoeDepth/blob/5b481547085d6cc26888f613fea37eab77fd7c8a/zoedepth/data/ibims.py#L60

That one is different, but interesting: it seems to transform float16 into 8bpc for depth image
https://github.com/isl-org/ZoeDepth/blob/5b481547085d6cc26888f613fea37eab77fd7c8a/zoedepth/data/ibims.py#L62

https://github.com/isl-org/ZoeDepth/blob/5b481547085d6cc26888f613fea37eab77fd7c8a/zoedepth/data/vkitti2.py#L152

https://github.com/isl-org/ZoeDepth/blob/5b481547085d6cc26888f613fea37eab77fd7c8a/zoedepth/data/diml_outdoor_test.py#L93-L95

EDIT: do you know why the code corresponding to each line is not appearing ? Is it because it references code from a different repo ? Or did I forget some kind of formatting flag ?

**Comment by thygate** [comment]
@AugmentedRealityCat  @graemeniedermayer 
Zoedepth now supported, by far best quality depthmaps, even without boost.

**Reply by AugmentedRealityCat** [reply]
It's Christmas again ! I knew it would make a big difference in quality as that's exactly what my early tests showed.
Thank you so much Bob.

**Reply by thygate** [reply]
I will add the mesh generation code from the zoedepth repo later too, it's a very clean and fast implementation that can also remove the occluded edges..

yes, wavefront obj ..

**Reply by AugmentedRealityCat** [reply]
I was really impressed by the 3d panorama feature as it's exactly what I was doing manually using your code for depth estimation and cinema4d for the 3d extraction, but just faster, cleaner and better. The 3d meshes produced with the Zoe-Depth code are also much cleaner, with none of the intersecting cross-triangulation problems I had with the 3d-photo-inpainting generated meshes. No need to remesh point clouds anymore to get proper topology for texture and lighting.

**Reply by thygate** [reply]
Yes the ZoeDepth output is quite amazing, definitely another step up. The mesh generation code is really beautiful and runs pretty fast too.
I'm including the spherical coordinate generation (equirectangular 360) too, I have not tried it yet ..

**Reply by AugmentedRealityCat** [reply]
One shortcoming of the current Zoe-Depth extension is that it doesn't do a seamless wrapping when creating 3d panoramas: there is a cut in the sphere, right at the frontier of the depthmap. 

This could be solved by generating depth maps that would tile seamlessly on X-axis, exactly like @ladypoly was asking for in another thread. I've done it manually in photoshop and it works.

**Reply by thygate** [reply]
Tiling support is on the list.

The mesh generation has been added.
